app/scripts/views/confirm.js

```diff
@@ -115,15 +121,22 @@ define(function (require, exports, module) {
               } else {
                 self.navigate('signup_complete');
               }
-            }, function (err) {
-              // The user's email may have bounced because it was invalid.
-              // Redirect them to the sign up page with an error notice.
-              if (AuthErrors.is(err, 'SIGNUP_EMAIL_BOUNCE')) {
-                self._bouncedEmailSignup();
-              } else {
-                self.displayError(err);
-              }
             });
+        }, function (err) {
+          // The user's email may have bounced because it was invalid.
+          // Redirect them to the sign up page with an error notice.
+          if (AuthErrors.is(err, 'SIGNUP_EMAIL_BOUNCE')) {
+            self._bouncedEmailSignup();
+          } else {
+            // Hide the error from the user, an error can happen
+            // here if the status api is overloaded or if the user is switching networks.
+            // Report errors to Sentry, but not the user.
+            // Details: github.com/mozilla/fxa-content-server/issues/2638.
+            self.sentryMetrics.captureException(err);
+            self.setTimeout(function () {
+              self._startPolling();
```

Comment:
Blanket re-starting the poll scares me a bit. Are there certain conditions we expect here where we know it's safe to restart? Are there conditions we know we should not restart?